apps/api/src/app/workflows-v2/mappers/workflow-template-get-mapper.ts

```diff
@@ -0,0 +1,77 @@
+import { IPreferenceChannels, StepTypeEnum, WorkflowOriginEnum } from '@novu/shared';
+import { DiscoverWorkflowOutputPreferences } from '@novu/framework';
+import { NotificationStepEntity, NotificationTemplateEntity } from '@novu/dal';
+
+import { MinifiedResponseWorkflowDto, StepResponseDto, WorkflowResponseDto } from '../dto/workflow.dto';
+
+export class WorkflowTemplateGetMapper {
+  static toResponseWorkflowDto(template: NotificationTemplateEntity): WorkflowResponseDto {
+    return {
+      _id: template._id,
+      tags: template.tags,
+      active: template.active,
+      critical: template.critical,
+      notificationGroupId: template._notificationGroupId,
+      preferences: WorkflowTemplateGetMapper.toPreferences(template.preferenceSettings), // Assuming this directly maps; may need adjustment
+      steps: template.steps.map(WorkflowTemplateGetMapper.toStepResponseDto),
+      name: template.name,
+      description: template.description,
+      origin: template.origin || WorkflowOriginEnum.NOVU,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+    };
+  }
+  static toMinifiedWorkflowDto(template: NotificationTemplateEntity): MinifiedResponseWorkflowDto {
+    return {
+      _id: template._id,
+      name: template.name,
+      tags: template.tags,
+      updatedAt: template.updatedAt || 'Missing Updated At',
+      stepSummery: template.steps.map(WorkflowTemplateGetMapper.buildStepSummery),
+    };
+  }
+
+  static toWorkflowsMinifiedDtos(templates: NotificationTemplateEntity[]): MinifiedResponseWorkflowDto[] {
+    return templates.map(WorkflowTemplateGetMapper.toMinifiedWorkflowDto);
+  }
+  static toStepResponseDto(step: NotificationStepEntity): StepResponseDto {
+    return {
+      name: step.name || 'Missing Name',
+      stepUuid: step._templateId,
```

Comment:
@LetItRock will that field name make sense to you in the FE? it should represent a Uniuqe identifier for a step within our system

Reply:
talked with @LetItRock until other pushback we are keeping stepUuid

Reply:
I'll take the opportunity to push back then 😛

This appears to be mapping a Mongo ID (not in UUID format) to a property named with `uuid`. That could be confusing for a developer who is expecting a specific format to be used. Could we simply go with `id` or `stepId`?

Reply:
I have no problem with that, but @djabarovgeorge said it would be confusing as the id without "_" as a prefix is usually being used in novu as an external id, kind of a unique name , wdyt? @rifont I have no real opinion here beside wanting to avoid have a "_id" is the dto, enything else is good for me